### PR TITLE
*: add a priority field to statement context and use it in store/tikv

### PR DESCRIPTION
**Pull request by tiancaiamao**
To support different priority, we need pass the priority information from SQL input to
underlying TiKV request.
As sugguested by @hanfei1991, priority is put in statement context to bypass plan builder.
Then it is passed to `distsql.Select` and store/tikv from statement context.

PTAL @hanfei1991 @zimulala

### CODE REVIEW

**Comment by hanfei1991** [comment]
LGTM

File: distsql/distsql.go [hunk 1]
```diff
@@ -184,7 +184,7 @@ func (pr *partialResult) Close() error {
 // concurrency: The max concurrency for underlying coprocessor request.
 // keepOrder: If the result should returned in key order. For example if we need keep data in order by
 //            scan index, we should set keepOrder to true.
```
Comment (zimulala) [inline distsql/distsql.go]: Add the comment about priority.

File: kv/kv.go [hunk 2]
```diff
@@ -147,6 +148,8 @@ type Request struct {
 	// ResponseIterator.Next is called. If concurrency is greater than 1, the request will be
 	// sent to multiple storage units concurrently.
 	Concurrency int
+	// Priority is the of this request.
```
Comment (zimulala) [inline kv/kv.go]: s/the of/the priority of ?

File: store/tikv/region_request.go [hunk 3]
```diff
@@ -72,7 +73,7 @@ func (s *RegionRequestSender) SendReq(bo *Backoffer, req *tikvrpc.Request, regio
 		}
 
 		s.storeAddr = ctx.Addr
```
Comment (zimulala) [inline store/tikv/region_request.go]: Why not set the priority here?

**Comment by tiancaiamao** [comment]
Distsql already support priority based on transaction size in TiKV, there is no need to pass hint information through the kv request, so I close this PR.

**Pull request closed without merging**